packages/tpetra/core/src/Tpetra_Details_unpackCrsMatrix.hpp

```diff
@@ -70,270 +72,345 @@ class Distributor;
 //
 namespace Details {
 
-template<class LocalMatrix, class LocalMap>
-typename LocalMatrix::ordinal_type
-combineCrsMatrixValues(LocalMatrix& lclMatrix,
-                       LocalMap& lclColMap,
-                       const typename LocalMatrix::ordinal_type lclRow,
-                       const typename LocalMatrix::ordinal_type numEnt,
-                       const typename LocalMatrix::value_type vals[],
-                       const typename LocalMap::global_ordinal_type cols[],
-                       const Tpetra::CombineMode combineMode,
-                       const bool atomic)
-{
-  typedef typename LocalMatrix::ordinal_type LO;
-
-  // INSERT doesn't make sense for a static graph, since you
-  // aren't allowed to change the structure of the graph.
-  // However, all the other combine modes work.
-
-  LO numValid = 0; // number of valid input column indices
-  if (combineMode == ADD) {
-    for (int k=0; k<numEnt; k++) {
-      LO lclColInd = lclColMap.getLocalElement(cols[k]);
-      numValid += lclMatrix.sumIntoValues(lclRow, &lclColInd, 1, &vals[k], false, atomic);
-    }
+/// \brief Reduction result for UnpackCrsMatrixAndCombineFunctor below.
+///
+/// The reduction result finds the offset and number of bytes associated with
+/// the first out of bounds error or unpacking error occurs.
+template<class LO>
+struct UnpackCrsMatrixError {
```

Comment:
*Not* addressed in 0d7d7ab.  I'll think about a unit test to test this and then implement it.